apps/mobile/src/app/_layout.tsx

```diff
@@ -41,6 +43,15 @@ initAppServices();
 initiateI18n();
 
 export default function RootLayout() {
+  const pathname = usePathname();
+  const params = useGlobalSearchParams();
+
+  useEffect(() => {
+    analytics.screen(pathname, {
+      params,
+    });
+  }, [pathname, params]);
```

Comment:
Even though a few LOC, this'd be great as its own hook with api like

`useOnRouteChange(route => analytics.screen(pathname, { params }))`